plugins/auth0/pkg/app/transform.go

```diff
@@ -2,23 +2,34 @@
 package app
 
 import (
+	"context"
 	"os"
+	"time"
 
 	"github.com/alecthomas/kong"
 	"github.com/aserto-dev/ds-load/sdk/plugin"
+	"github.com/aserto-dev/ds-load/sdk/transform"
 )
 
 type TransformCmd struct {
-	Template     string `name:"template" short:"t" env:"DS_TEMPLATE_FILE" help:"transformation template file path" type:"path" optional:""`
-	MaxChunkSize int    `name:"max-chunk-size" env:"DS_MAX_CHUNK_SIZE" help:"maximum chunk size" default:"20" optional:""`
+	Template string `name:"template" short:"t" env:"DS_TEMPLATE_FILE" help:"transformation template file path" type:"path" optional:""`
 }
 
-func (t *TransformCmd) Run(context *kong.Context) error {
-	content, err := t.getTemplateContent()
+func (t *TransformCmd) Run(kongContext *kong.Context) error {
+	template, err := t.getTemplateContent()
 	if err != nil {
 		return err
 	}
-	return plugin.NewDSPlugin(plugin.WithTemplate(content), plugin.WithMaxChunkSize(t.MaxChunkSize)).Transform()
+
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), 1500*time.Millisecond)
+	defer cancel()
+
+	goTemplateTransformer := transform.NewGoTemplateTransform(template)
```

Comment:
To make it absolutely clear that we're binding to the Transformer interface I'd define the variable explicitly:
```go
var transformer Transformer
transformer = transform.NewGoTemplateTransform(template)
return transformer.Transform(timeoutCtx, os.Stdin, os.Stdout, os.Stderr)
```

And the same goes for other places we create a transformer.

Reply:
In fact, perhaps the plugin SDK should explicitly provide a way for a plugin implementation to register a factory function for its fetcher and transformer.
This way we won't have to duplicate the implementation of `TranformCmd` between all the plugins. Right now that code is copied and pasted everywhere.
If the purpose of the plugin SDK is to simplify the development of plugins, reducing that boilerplate would reduce the code substantially.